hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -2444,8 +2451,16 @@ public FlushResultImpl flushcache(boolean forceFlushAllStores, boolean writeFlus
       }
 
       try {
-        Collection<HStore> specificStoresToFlush =
+        Collection<HStore> specificStoresToFlush = null;
+        if (!forceFlushAllStores && families != null) {
+          specificStoresToFlush = stores.entrySet().stream()
+            .filter(e -> families.contains(Bytes.toString(e.getKey())))
```

Comment:
If we want to use contains then we better use a Set instead of a List?

Reply:
> So why we choose String instead of byte[] here?

1、The family name stored as private byte[] in ImmutableByteArray, if use byte[], it need twice convertion.
2、We use the method of contains to filter them later, use string seems esaliy to do.
Correct me if i have some mistake.

> If we want to use contains then we better use a Set instead of a List?
Yeah, you are right, wil fix.

Thanks a lot.

Reply:
If the String is ok, i will push a commit later, or else i will research more.
Thanks.

Reply:
In the WAL interface, we all use byte[] as family name, so I prefer we also use byte[] here. You could use TreeSet with BytesComparator to hold byte[].

Reply:
Ok, will use byte[] instead of String.
In addition,since we does not need a sorted collection,  i could iterate the families to avoid use contains.
Thanks.